### PR TITLE
refactor: migrate html adapter to extension

### PR DESCRIPTION
**Pull request by donteatfriedrice**
[BS-1852](https://linear.app/affine-design/issue/BS-1852/migrate-html-adapter)

### CODE REVIEW

**Comment by donteatfriedrice** [comment]
* **#8868** <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/8868?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/8868?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `master`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/affine/shared/src/adapters/markdown/delta-converter.ts [hunk 1]
```diff
@@ -101,6 +101,7 @@ export class MarkdownDeltaConverter extends DeltaASTConverter<
   astToDelta(ast: MarkdownAST): DeltaInsert<AffineTextAttributes>[] {
     const context = {
       configs: this.configs,
+      options: {},
```
Comment (fourdim) [inline packages/affine/shared/src/adapters/markdown/delta-converter.ts:104]: `Object.create(null);` or `Maps`

File: packages/affine/shared/src/adapters/plain-text/delta-converter.ts [hunk 2]
```diff
@@ -36,6 +36,7 @@ export class PlainTextDeltaConverter extends DeltaASTConverter<
   astToDelta(ast: string) {
     const context = {
       configs: this.configs,
+      options: {},
```
Comment (fourdim) [inline packages/affine/shared/src/adapters/plain-text/delta-converter.ts:39]: Same here.

File: packages/affine/shared/src/adapters/html-adapter/delta-converter.ts [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        context.current = mdast;
+      }
+    }
+
+    return mdast;
+  }
+
+  private _spreadAstToDelta(
+    ast: HtmlAST,
+    options: DeltaASTConverterOptions = {}
```
Comment (fourdim) [inline packages/affine/shared/src/adapters/html-adapter/delta-converter.ts:101]: Same here.

File: packages/affine/shared/src/adapters/html-adapter/delta-converter.ts [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+      }
+    }
+    return 'children' in ast
+      ? ast.children.flatMap(child => this._spreadAstToDelta(child, options))
+      : [];
+  }
+
+  astToDelta(
+    ast: HtmlAST,
+    options: DeltaASTConverterOptions = {}
```
Comment (fourdim) [inline packages/affine/shared/src/adapters/html-adapter/delta-converter.ts:121]: Same here.

File: packages/affine/shared/src/adapters/utils/hast.ts [hunk 5]
```diff
@@ -274,3 +271,15 @@ export const hastIsParagraphLike = (node: Element): boolean => {
...
+export const HastUtils = {
+  isElement,
+  hastGetTextContent,
+  hastGetElementChildren,
+  hastGetTextChildren,
+  hastGetTextChildrenOnlyAst,
+  hastGetInlineOnlyElementAST,
+  hastQuerySelector,
+  hastFlatNodes,
+  hastIsParagraphLike,
```
Comment (fourdim) [inline packages/affine/shared/src/adapters/utils/hast.ts:284]: Remove the hast prefix.

File: packages/blocks/src/_common/adapters/html-adapter/index.ts
Comment (fourdim) [inline packages/blocks/src/_common/adapters/html-adapter/index.ts:1]: Empty file.

File: packages/playground/apps/_common/components/debug-menu.ts [hunk 6]
```diff
@@ -161,6 +160,23 @@ function getDarkModeConfig(): boolean {
...
+}
+
+type AdapterFactoryIdentifier =
+  | typeof HtmlAdapterFactoryIdentifier
+  | typeof MarkdownAdapterFactoryIdentifier
+  | typeof PlainTextAdapterFactoryIdentifier;
+
+interface AdapterConfig {
+  identifier: AdapterFactoryIdentifier;
+  fileExtension: string;
```
Comment (fourdim) [inline packages/playground/apps/_common/components/debug-menu.ts:175]: Add comments here to mention that the `fileExtension` needs a dot

**Review by fourdim — APPROVED** [review]

**Pull request merged**